contrib/coredns/policy/policy.go

```diff
@@ -119,11 +120,39 @@ func (p *PolicyMiddleware) getEDNS0Attrs(r *dns.Msg) ([]*pb.Attribute, bool) {
 	return attrs, foundSourceIP
 }
 
+// This function validates the middleware response
+// Inputs is *PolicyMiddle, *Response and error
+// Output is DNS status and error
+func ValidateMiddlewareResponse(p *PolicyMiddleware, w dns.ResponseWriter, r *dns.Msg,response *Response, status int, err error) (int, error) {
```

Comment:
Had moved it as an idempotent function so validation was for it only. Pulling code into ServeDns.